### PR TITLE
Allow a custom CaptionVariableReplacementHandler

### PR DESCRIPTION
**Pull request by 0utplay**
This PR adds getter and setter to the CommandManager,
making it possible to set the CaptionVariableReplacementHandler ourself.

### CODE REVIEW

File: cloud-core/src/main/java/cloud/commandframework/CommandManager.java [hunk 1]
```diff
@@ -240,6 +243,26 @@ protected CommandManager(
...
+        return this.captionVariableReplacementHandler;
+    }
+
+    /**
+     * Set the caption variable replacement handler
+     *
+     * @param captionVariableReplacementHandler new replacement handler
+     */
+    public void setCaptionVariableReplacementHandler(
+            @NonNull final CaptionVariableReplacementHandler captionVariableReplacementHandler
```
Comment (jpenilla) [inline cloud-core/src/main/java/cloud/commandframework/CommandManager.java:261]: `final @NonNull`

File: cloud-core/src/main/java/cloud/commandframework/CommandManager.java [hunk 2]
```diff
@@ -240,6 +243,26 @@ protected CommandManager(
...
+    public @NonNull CaptionVariableReplacementHandler getCaptionVariableReplacementHandler() {
+        return this.captionVariableReplacementHandler;
+    }
+
+    /**
+     * Set the caption variable replacement handler
+     *
+     * @param captionVariableReplacementHandler new replacement handler
+     */
+    public void setCaptionVariableReplacementHandler(
```
Comment (jpenilla) [inline cloud-core/src/main/java/cloud/commandframework/CommandManager.java:260]: no set prefix

File: cloud-core/src/main/java/cloud/commandframework/CommandManager.java [hunk 3]
```diff
@@ -240,6 +243,26 @@ protected CommandManager(
         return this.command(command.manager(this).build());
     }
 
+    /**
+     * Get the caption variable replacement handler
+     *
+     * @return the caption variable replacement handler
+     */
+    public @NonNull CaptionVariableReplacementHandler getCaptionVariableReplacementHandler() {
```
Comment (jpenilla) [inline cloud-core/src/main/java/cloud/commandframework/CommandManager.java:251]: no get prefix

File: cloud-core/src/main/java/cloud/commandframework/CommandManager.java [hunk 4]
```diff
@@ -240,6 +243,26 @@ protected CommandManager(
...
+        return this.captionVariableReplacementHandler;
+    }
+
+    /**
+     * Set the caption variable replacement handler
```
Comment (jpenilla) [inline cloud-core/src/main/java/cloud/commandframework/CommandManager.java:256]:
```suggestion
     * Sets the caption variable replacement handler.
```

**Review by jpenilla — CHANGES_REQUESTED** [review]
Needs to update changelog

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+
+### Changed
```
Comment (jpenilla) [inline CHANGELOG.md:9]:
```suggestion
### Added
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
...
+- The command manager specifies the caption replacement handler
```
Comment (jpenilla) [inline CHANGELOG.md:10]:
```suggestion
- Allow for setting a custom `CaptionVariableReplacementHandler` on the command manager ([#352](https://github.com/Incendo/cloud/pull/352))
```

**Review by jpenilla — CHANGES_REQUESTED** [review]

**Comment by 0utplay** [comment]
Hey, is something wrong with the PR or not up to your standards?

**Comment by Citymonstret** [comment]
I think this is a sensible contribution. It has some merge conflicts that would currently prevent it from being merged. But I don't have any objections to this being added to v1.7.

**Review by Citymonstret — APPROVED** [review]

**Pull request merged**